src/swsssdk/dbconnector.py

```diff
@@ -17,23 +18,29 @@ def __init__(self, **kwargs):
 
 SonicV1Connector.db_map = _connector_map[SonicV1Connector.__name__]['db_map']
 
-if len(SonicV1Connector.db_map) != len({v['db'] for k, v in SonicV1Connector.db_map.items()}):
-    raise RuntimeError("Duplicate DB index detected in configuration.")
-
-
 class SonicV2Connector(DBInterface):
     def __init__(self, **kwargs):
         super(SonicV2Connector, self).__init__(**kwargs)
 
+    # after py test cases(e.g snmpagent/tests) are modified
+    # we can do the replacement on this new API via renaming connect_new to connect
+    # def connect(self, db_name, retry_on=True, isTcpconn=False):
+    def connect_new(self, db_name, retry_on=True, isTcpconn=False):
```

Comment:
>connect_new [](start = 8, length = 11)

Is it possible to keep original function and change its implementation?

Reply:
The original connect() is from DBInterface class, which is not modified. Here I just overwrite the connect() in SonicV2Connector which is the child class to prepare some parameters, finally it is still call the original connect in DBInterface. now it is named as connect_new for temporally, since we didn't fix the py test cases yet which will cause test failed. Finally we will use name connect() which is the original name.

Reply:
use connect() instead